client/session.go

```diff
@@ -1774,12 +1780,20 @@ func (s *session) streamBlocksMetadataFromPeerV2(
 
 		for _, elem := range result.Elements {
 			blockStart := time.Unix(0, elem.Start)
+
+			// Reset the reused ID data
+			reusedIDBytes.Resize(0)
+			reusedIDBytes.AppendAll(elem.ID)
+
+			// Clone the ID
+			clonedID := idPool.Clone(reusedID)
+
```

Comment:
Hey to clean up this code we should probably directly use the pool from `resultOpts`.

e.g.
```go
bytesPool := resultOpts.DatabaseBlockOptions().BytesPool()
```